### ISSUE TITLE
cannot run code on php interpretuere working in docker

### ISSUE THREAD

**Issue by Taalaybek** [Roboroads/laravel-tinker#46]
**Bug description**
shows a progress but not output result in right window of tinker

**To Reproduce**
when running code via docker, it is just try up container that I set in settings of php interpretuere

**Environment (please complete the following information):**
 - IDE & Version: PHPStorm 2020.2.4
 - PHP Version: 7.4
 - OS: Elementary OS 5..1.7 Hera (based on Ubuntu 18.04)
 - Laravel Tinker plugin version : [e.g. 1.1.1]
 - Interpreter TypeDocker / Docker-compose
 - Laravel Version: 8.0
 - Tinker version: 2.0

**Comment by Roboroads** [comment]
@Taalaybek Can you confirm *how* you are running docker? If you are using `docker run` can you try is the issue persists if you use `docker exec`?

I've had another problem which I can't seem to figure out using `docker run`.

**Comment by dmitryuk** [comment]
I'm using docker-compose, running tests from PHPStorm works fine, Laravel tinker 2 - no.

**Comment by Roboroads** [comment]
@dmitryuk That does not really answer my question yet. Are you using `docker-compose run` or `docker-compose exec` when running Tinker? You can find it in the PHPStorm interpreter settings.

**Comment by dmitryuk** [comment]
`docker-compose exec`
![image](https://user-images.githubusercontent.com/661654/109057280-204e8480-7714-11eb-9634-5b15e4611a66.png)

**Comment by hailwood** [comment]
On this topic [Lando](https://docs.lando.dev/config/laravel.html) is quite popular,
The issue is that it's a wrapper around docker-compose - so we don't have the docker-compose.yml file

Instead we do things like `lando artisan migrate` or `lando tinker` and it will deal with that side for us.
It means we cannot use PHPstorm for running tests etc 

I doubt this is something this plugin can solve, but it does prevent us from using it.

**Comment by Roboroads** [comment]
@hailwood True, i'd have to create a switch between "how to run a tinker command" with you  being able to add a custom command for executing. If there is enough people wanting that I could add it.

**Comment by jdecode** [comment]
If you need time/energy/effort to work on this one, I can try to chip in...

**Comment by Roboroads** [comment]
@hailwood Thanks for noting. Are you actually having a bug that prevents you from running with Docker, or do you need a way to change the command? If it''s the latter, this is a feature request and not a bug and you should make a new ticket. If it';s the first, please provide the docker container/dockerfile you are trying to run it on :blush:

**Comment by minhnh94** [comment]
![CleanShot 2022-01-04 at 12 07 13](https://user-images.githubusercontent.com/6817741/148004395-2d0a4d31-383f-4954-9dc9-f9652f043910.png)
Have the same issue, with PHP interpreter coming from docker container. tinker executed but no output. If I change to docker-compose exec it doesn't even run at all.
Which is weird, the exact same setting worked before and I never touched it since then, I wonder if an update on Laravel tinker plugin broke it ... 😞

**Comment by Warmic** [comment]
Hello, I encountered the same problem and fixed it by downgrading PhpStorm version from 2021.3.1 to 2021.2.4. I also tried 2021.3.2 (EAP) it too had this problem

**Comment by Warmic** [comment]
Hey guys,I think I understand now what is going on. 
When IDE tries to execute command in fresh docker container, it does following: docker-compose -f "interpreter.docker-compose.yml" -f "ide-created/docker-compose.override.yml" run --rm containerName

This docker-compose.override file contains php code that will be executed. Both from plugin and user-input. Problem happens here. In versions of IDE where plugin works correctly, all variables (e.g. with $ sign) get prefixed by $ twice(so instead of $var = 1, it becomes $$var = 1). Then this code gets executed and works fine. In 2021.3.1 version, however, it gets prefixed 4 times instead of 2. And execution of this code results in nothing, because it fails before it even gets to php. If I replace all 4 consecutive $ by 2, code is executed properly!

Now, I tried to fix it by myself. But unfortunately, was unable to understand which IDE-specific files I should modify. 
@Roboroads do you have any thoughts on how it can be fixed?

Here are docker-compose files I was talking about.
<img width="748" alt="invalid compose" src="https://user-images.githubusercontent.com/22577007/151757227-9a2ef2b3-a4df-4ab1-8cac-59819c6d1883.png">
<img width="748" alt="valid compose" src="https://user-images.githubusercontent.com/22577007/151757233-d12c24a5-76b4-4b4a-a3cc-f5dafca0ad3c.png">

**Comment by Warmic** [comment]
@Roboroads if you don't know how to solve this problem, I can try to contact some guys from Jetbrains or make an issue on their youtrack page

**Comment by JerryBels** [comment]
Hey, in a sail environment on Windows I seem to have a similar issue. Tinker is executed but no output shown and the command actually has no effect (adding something in DB for example doesn't work). I configured my cli interpreter to use docker, but to no avail.

**Comment by Roboroads** [comment]
This PR got me thinking of something. I'll keep everyone posted.

**Comment by Roboroads** [comment]
@Warmic I posted this issue and your findings in their slack channel. I have no idea how to het this overrides dockerfile out so if you have time to explain it to me I can maybe add a bug to youtrack myself :) (or you do it, and I back you with "My plugin has this issue", also good :smile: ).

**Comment by Roboroads** [comment]
JB sais it's fixed in PHPStorm 2022.2 EAP Can someone also test it?

**Comment by peresmishnyk** [comment]
@Roboroads Testing PHPStorm 2022.2 EAP, not fixed :(

**Comment by Roboroads** [comment]
I'm one step further, I'm certain it has something to do with functions. I rewrote tinker_run.php without declaring functions and it works.

**Comment by Roboroads** [comment]
</proffessional>
I. FKN. FIXED. THIS. FINALLY.
FKN. INTELLIJ. WEIRDNESS.
<proffessional>

Ahum. Release is on its way :)

**Comment by Roboroads** [comment]
Fix is ready and will be released in 2.3.4.

Please let me know if you still have problems, but I hope I can close this one once and for all. It took be more than 40 hours to debug this :D

**Comment by Roboroads** [comment]
Before reopening this bug again, please [read this](https://github.com/Roboroads/laravel-tinker/issues/167)

**Comment by Roboroads** [comment]
@JohnLeet Add me on Discord - same name as github, we'll look at it together :)
